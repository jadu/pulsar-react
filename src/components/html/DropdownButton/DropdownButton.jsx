import React from 'react';
import classnames from 'classnames';
import ButtonGroup from '../ButtonGroup/ButtonGroup';
import Button from '../Button/Button';

export default class DropdownButton extends React.Component {

  static defaultProps = {
    'aria-expanded': 'false',
    items: [],
    label: 'Label'
  };

  render() {
    let {
      children,
      className,
      danger,
      info,
      inverse,
      items,
      label,
      naked,
      outline,
      primary,
      success,
      warning,
      white,
      ...props
    } = this.props;

    let variantClasses = classnames(className, 'btn dropdown__toggle', {
      'btn--danger': danger,
      'btn--info': info,
      'btn--inverse': inverse,
      'btn--naked': naked,
      'btn--outline': outline,
      'btn--primary': primary,
      'btn--success': success,
      'btn--warning': warning,
      'btn--white': white
    });

    return (
      <ButtonGroup className="dropdown">
        <Button
          className={variantClasses}
          {...props}
        >
          {label} <span className="caret"></span>
        </Button>
        <ul className="dropdown__menu pull-left">
          {
            items.map(function(item, i) {
              let {label, ...props} = item;
              props.key = i;

              return <li {...props}>{label}</li>
            })
          }
        </ul>
      </ButtonGroup>
    );
  }
}


// <div  class="btn__group dropdown">
//   <button aria-haspopup="true" aria-expanded="false" aria-controls="guid-1195386946" data-toggle="dropdown"  class="btn dropdown__toggle">
//     My Action&nbsp;<span class="caret"></span>
//   </button>
//   <ul id="guid-1195386946"  class="dropdown__menu pull-left">
//     <li><a href="#foo">Foo</a></li>
//     <li><a href="#bar">Bar</a></li>
//   </ul>
// </div>