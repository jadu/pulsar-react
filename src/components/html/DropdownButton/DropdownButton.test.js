'use strict';

import React from 'react';
import { shallow, render } from 'enzyme';

import DropdownButton from './DropdownButton';

it('renders the basic dropdown button', () => {
  const tree = shallow(
    <DropdownButton />
  );
  expect(tree).toMatchSnapshot();
});

it('allows a custom label', () => {
  const tree = shallow(
    <DropdownButton label="Hello" />
  );
  expect(tree).toMatchSnapshot();
});

it('add additional classes on the button', () => {
  const tree = shallow(
    <DropdownButton 
      className="foo"
      label="Hello" />
  );
  expect(tree).toMatchSnapshot();
});

it('renders one list item', () => {
  const tree = shallow(
    <DropdownButton 
      items={[
        { id: 'chocolate', label: 'Chocolate' }
      ]}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('renders multiple list items', () => {
  const tree = shallow(
    <DropdownButton 
      items={[
        { id: 'chocolate', label: 'Chocolate' },
        { id: 'strawberry', label: 'Strawberry' },
        { id: 'vanilla', label: 'Vanilla' }
      ]}
    />
  );
  expect(tree).toMatchSnapshot();
});

it('should obey the button variations', () => {
  const tree = render(
    <DropdownButton primary />
  );
  expect(tree).toMatchSnapshot();
});
