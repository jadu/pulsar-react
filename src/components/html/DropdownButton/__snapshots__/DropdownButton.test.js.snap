// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add additional classes on the button 1`] = `
<ButtonGroup
  className="dropdown"
>
  <Button
    aria-expanded="false"
    className="foo btn dropdown__toggle"
    tag="button"
  >
    Hello
     
    <span
      className="caret"
    />
  </Button>
  <ul
    className="dropdown__menu pull-left"
  />
</ButtonGroup>
`;

exports[`allows a custom label 1`] = `
<ButtonGroup
  className="dropdown"
>
  <Button
    aria-expanded="false"
    className="btn dropdown__toggle"
    tag="button"
  >
    Hello
     
    <span
      className="caret"
    />
  </Button>
  <ul
    className="dropdown__menu pull-left"
  />
</ButtonGroup>
`;

exports[`renders multiple list items 1`] = `
<ButtonGroup
  className="dropdown"
>
  <Button
    aria-expanded="false"
    className="btn dropdown__toggle"
    tag="button"
  >
    Label
     
    <span
      className="caret"
    />
  </Button>
  <ul
    className="dropdown__menu pull-left"
  >
    <li
      id="chocolate"
      key="0"
    >
      Chocolate
    </li>
    <li
      id="strawberry"
      key="1"
    >
      Strawberry
    </li>
    <li
      id="vanilla"
      key="2"
    >
      Vanilla
    </li>
  </ul>
</ButtonGroup>
`;

exports[`renders one list item 1`] = `
<ButtonGroup
  className="dropdown"
>
  <Button
    aria-expanded="false"
    className="btn dropdown__toggle"
    tag="button"
  >
    Label
     
    <span
      className="caret"
    />
  </Button>
  <ul
    className="dropdown__menu pull-left"
  >
    <li
      id="chocolate"
      key="0"
    >
      Chocolate
    </li>
  </ul>
</ButtonGroup>
`;

exports[`renders the basic dropdown button 1`] = `
<ButtonGroup
  className="dropdown"
>
  <Button
    aria-expanded="false"
    className="btn dropdown__toggle"
    tag="button"
  >
    Label
     
    <span
      className="caret"
    />
  </Button>
  <ul
    className="dropdown__menu pull-left"
  />
</ButtonGroup>
`;

exports[`should obey the button variations 1`] = `
<div
  class="dropdown"
>
  <button
    aria-expanded="false"
    class="btn dropdown__toggle btn--primary"
  >
    Label 
    <span
      class="caret"
    />
  </button>
  <ul
    class="dropdown__menu pull-left"
  />
</div>
`;
